### PR TITLE
PER-447 - Provisioning EKS in EDB-Terraform

### PR DESCRIPTION
**Pull request by dougortiz**
- Adds support for AWS Elastic Kubernetes Service
- Updated 'README.md' to reflect Azure Kubernetes Service and AWS Elastic Kubernetes Service

### CODE REVIEW

File: edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  description = "Endpoint for EKS control plane"
+  value       = module.eks.cluster_endpoint
+}
+
+output "cluster_security_group_id" {
+  description = "Security group ids attached to the cluster control plane"
+  value       = module.eks.cluster_security_group_id
+}
+
+output "cluster_certificate" {
```
Comment (bryan-bar) [inline edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf:16]: This output is named `ca_certification` under google's gke. This should be updated to either match googles outputs or googles outputs should be changed to match eks.

https://github.com/EnterpriseDB/edb-terraform/blob/e9f4504d27f4c7ddd688422022d845d7ce5bea0b/edbterraform/data/terraform/azure/modules/kubernetes/outputs.tf#L11

File: edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,30 @@
+output "cluster_id" {
+  description = "EKS cluster ID"
+  value       = module.eks.cluster_id
+}
+
+output "cluster_endpoint" {
```
Comment (bryan-bar) [inline edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf:6]: This is named `host` for azure and google. This should be updated so all provider outputs match where possible.

https://github.com/EnterpriseDB/edb-terraform/blob/e9f4504d27f4c7ddd688422022d845d7ce5bea0b/edbterraform/data/terraform/azure/modules/kubernetes/outputs.tf#L26-L28

https://github.com/EnterpriseDB/edb-terraform/blob/e9f4504d27f4c7ddd688422022d845d7ce5bea0b/edbterraform/data/terraform/gcloud/modules/kubernetes/outputs.tf#L8-L11

File: infrastructure-examples/aws-eks.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+aws:
+  tags:
+    cluster_name: EDB-K8s-CNP
+    created_by: edb-terraform
+  ssh_user: rocky
+  operating_system:
+    name: Rocky-8-ec2-8.6-20220515.0.x86_64
+    owner: 679593333241
```
Comment (bryan-bar) [inline infrastructure-examples/aws-eks.yml:8]: This can be removed as it is not used.

**Review by bryan-bar — CHANGES_REQUESTED** [review]
@dougortiz 
I was not able to test on my machine due to permissions, but we went over that issue last time and tested it together. If you could test one more time after making the changes requested and it should be good to go.

If you know which permissions are needed, if you could create a `README` under the kubernetes module for aws (eks). If not, we could come back to it since it will be needed at some point for all the modules, for each provider.

File: edbterraform/data/templates/aws/kubernetes.tf.j2 [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    aws = aws.{{ region_ }}
+    kubernetes = kubernetes.{{ region_ }}
+  }
+
+# This statement works for: Azure Kubernetes Services and Google Kubernetes Engine
+# TODO: Future versions of AWS Elastic Kubernetes Services might support it.
+#       Check on any updates that occur to AWS EKS Modules versions.
+#       This bug causes 'terraform plan' to fail since it forces aws k8s modules variable casting issues
+#       Refer to: https://support.hashicorp.com/hc/en-us/articles/4408936406803-Kubernetes-Provider-block-fails-with-connect-connection-refused-
+#  depends_on = [ null_resource.validation ]
```
[the remark this answers is not in the document]
Comment (bryan-bar) [reply edbterraform/data/templates/aws/kubernetes.tf.j2:21]: There is also support for us to make our own set of modules, if need be:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster

**Comment by bryan-bar** [comment]
Note: the Kubernetes VPC's from each provider is not connected with region peering at the moment and this will eventually need to be handled so that we do not need to use the public interface, vpn or kubectl when using virtual machines, or other, to simulate workloads within a provider's private network.

File: edbterraform/data/templates/aws/main.tf.j2 [hunk 5]
```diff
@@ -10,9 +10,11 @@
 {%     include "network.tf.j2" %}
 {%   endif %}
 
-{%   if has_machines %}
+{%   if has_machines or has_kubernetes %}
 {%     include "key_pair.tf.j2" %}
+{%   endif %}
 
+{%   if has_machines %}
```
[the remark this answers is not in the document]
Comment (dougortiz) [reply edbterraform/data/templates/aws/main.tf.j2:17]: Reverted to original code

File: edbterraform/data/templates/aws/kubernetes.tf.j2 (hunk 4 above)
[the remark this answers is not in the document]
Comment (dougortiz) [reply edbterraform/data/templates/aws/kubernetes.tf.j2:21]: Added link: https://github.com/hashicorp/terraform/issues/30937

File: edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf (hunk 1 above)
Comment (dougortiz) [reply edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf:16]: Updated to ca_certification

File: edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf (hunk 2 above)
Comment (dougortiz) [reply edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf:6]: Updated to "host"

File: infrastructure-examples/aws-eks.yml (hunk 3 above)
Comment (dougortiz) [reply infrastructure-examples/aws-eks.yml:8]: Removed all 3 lines

File: edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf (hunk 1 above)
Comment (bryan-bar) [reply edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf]: @dougortiz if you could update this to `ca_certificate` to match azure's aks outputs. 

I referenced `ca_certification` and google's gke in the last review but that was a typo on my part and mentioned the wrong provider. Link below shows what azure's aks output has.

https://github.com/EnterpriseDB/edb-terraform/blob/e9f4504d27f4c7ddd688422022d845d7ce5bea0b/edbterraform/data/terraform/azure/modules/kubernetes/outputs.tf#L11

Comment (dougortiz) [reply edbterraform/data/terraform/aws/modules/kubernetes/outputs.tf]: Renamed and updated to: 'ca_certificate' in outputs.tf

**Review by bryan-bar — APPROVED** [review]
👍

**Pull request merged**